src/nnue/nnue_common.h

```diff
@@ -159,6 +159,48 @@ namespace Stockfish::Eval::NNUE {
               write_little_endian<IntType>(stream, values[i]);
   }
 
+  template <typename IntType>
+  inline IntType read_leb_128(std::istream& stream) {
+      static_assert(std::is_signed_v<IntType>, "Not implemented for unsigned types");
+      IntType result = 0, shift = 0;
+      while (true) {
+          std::uint8_t byte;
+          stream.read(reinterpret_cast<char*>(&byte), sizeof(std::uint8_t));
```

Comment:
read error should be handled somehow

Reply:
Should it also be done in `read_little_endian` and `write_little_endian`?

Reply:
It's not necessary there. Here it's needed to prevent an infinite loop. Other mitigations could also suffice.

Reply:
also, note that shifting by more than the type width is technically bad, so we should avoid that

Reply:
Avoided shifting by more than the type now.